qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -289,27 +303,29 @@ def observable(me_table, init_term=0, mapping="jordan_wigner", wires=None):
 
     sp_op_shape = (3,)
     tp_op_shape = (5,)
-    for i_table in me_table:
-        if np.array(i_table).shape not in (sp_op_shape, tp_op_shape):
-            raise ValueError(
-                "expected entries of 'me_table' to be of shape (3,) or (5,) ; got {}".format(
-                    np.array(i_table).shape
+    for table in me_tables:
+        for row in table:
+            if np.array(row).shape not in (sp_op_shape, tp_op_shape):
+                raise ValueError(
+                    "Expected entries of matrix element tables to be of shape (3,) or (5,); got {}".format(
+                        np.array(row).shape
+                    )
```

Comment:
Could we have
```python
shape = table.shape

if not (len(shape) == 2 and shape[1] in [3, 5]):
   raise ...
```
Was just wondering, since the current approach allows for `table` to be a list of 1D numpy arrays of different size, and I'm not sure if that makes sense in this setting, i.e., I thought each `table` had to be a 2-dimensional array.

Reply:
Checking the shape is a good point @trbromley, thanks. Will add this.